test/BerfChatStorage.ts

```diff
@@ -175,12 +160,68 @@ describe("BerfChatStorage contract tests", async () => {
             EthCrypto.hash.keccak256(decryptedPayload.message)
         );
 
-        console.log(
-            'Got message from ' +
-            senderAddress +
-            ': ' +
-            decryptedPayload.message
+        expect(senderAddress).to.equal(accountUno.address);
+
+        expect(decryptedPayload.message).to.equal(secretMessage);
+    })
+    */
+
+    it("confirms that only the public key's private key can decrypt the message", async () => {
+        // Create new accounts
+        // (Used the EthCrypto library to create accounts
+        // rather than using Hardhat because for testing
+        // purposes, it is much easier to retrieve 
+        // public keys with it.)
+        const accountUno = EthCrypto.createIdentity();
+        const accountDos = EthCrypto.createIdentity();
+        const accountTres = EthCrypto.createIdentity();
+
+        // Declare and assign variable
+        // with secret string.
+        const secretMessage = "I know who Satoshi is.";
+
+        // Sign the message with accountUno's
+        // private key
+        const signature = EthCrypto.sign(
+            accountUno.privateKey,
+            EthCrypto.hash.keccak256(secretMessage)
         );
+
+        // Create object with message and signature
+        const payload = {
+            message: secretMessage,
+            signature
+        };
+
+        // Encrypt the message and the signature with
+        // accountDos' publicKey. (By encrypting it
+        // with accountDos' public key, only accountDos
+        // can decrypt `payload` with his private key.)
+        const encrypted = await EthCrypto.encryptWithPublicKey(
+            accountDos.publicKey,
+            JSON.stringify(payload) // Stringify the payload to allow encryption
+        );
+
+        // Convert object into smaller string-representation
+        const encryptedString = EthCrypto.cipher.stringify(encrypted);
+
+
+        // ~~~~ SENDING MESSAGE FROM accountOne TO accountTwo ~~~ //
+
+
+        // Parse the string back to an object
+        const encryptedObject = EthCrypto.cipher.parse(encryptedString);
+
+        // Attempt to decrypt with accountTres,
+        // instead of accountDos
+        /*
+        return expect(Promise.resolve(
+            await EthCrypto.decryptWithPrivateKey(
+                accountTres.privateKey,
+                encryptedObject
+            )
+        )).to.eventually.throw("Error: Bad MAC");
         */
-    })
+       expect(function () { EthCrypto.decryptWithPrivateKey(accountTres.privateKey, encryptedObject); }).to.throw();
```

Comment:
expect(await EthCrypto.decryptWithPrivateKey(accountTres.privateKey, encryptedObject)).to.throw()

Reply:
resolved with `await rejects()`